session_test.go

```diff
@@ -1254,3 +1255,30 @@ func TestSession_ConnectionWriteTimeout(t *testing.T) {
 
 	wg.Wait()
 }
+
+func TestCancelAccept(t *testing.T) {
+	_, server := testClientServer()
+	defer server.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+
+		stream, err := server.AcceptStreamWithContext(ctx)
+		if err != context.Canceled {
+			t.Fatalf("err: %v", err)
```

Comment:
`t.Fatal` isnt' intended to be called in goroutines (as it calls `t.FatalNow` [1]).  I would suggest setting the error to a variable in function scope and asserting in main goroutine or using a channel to send error instead of using the waitGroup.

[1] https://golang.org/pkg/testing/#T.FailNow

Reply:
I followed the code used elsewhere. If it should be changed, we should do it for all the tests.